samples/nrf9160/modem_shell/src/gnss/gnss.c

```diff
@@ -133,17 +134,44 @@ static uint8_t pvt_output_level = 2;
 static uint8_t nmea_output_level;
 static uint8_t event_output_level;
 
+static const char *gnss_err_to_str(int err)
```

Comment:
Because this is not GNSS specific, it raises questions on whether we should do this more within MoSH, or not do even here?

Reply:
Yes, this is generic and should probably be somewhere else. Anyway, I would like to print a readable reason for the error, because it makes life so much easier, so at least I don't want to just remove this mapping.

Reply:
Personally, I'm not strongly for or against having this functionality in the first place. It's nice but on the other hand, it's fairly standard to just print the error code.

We have at least 265 occurrences of printing `err` or `ret` in MoSH and probably more that I wasn't able to grep. You probably don't want to go through all of them but I guess this should be within utils so it could be used elsewhere.

Then the question is, should the function provide directly "%d (%s)" type of string.
And if this is general, I guess this should rather print ENOMEM, not NRF_ENOMEM.

Reply:
The reason why I wanted to have it here is, that it's easy to get errors for GNSS commands. For example, here's your favorite case:
```
mosh:~$ gnss start
GNSS: Failed to start GNSS, error: -13 (NRF_EACCES)
```
And if you look at `nrf_modem_gnss.h`:
```
 * @retval -NRF_EACCES GNSS is not enabled in system or functional mode.
```
Most of the GNSS commands are such, that you can't execute them in any state. Earlier there was a state check in mosh for this, but this prevented us from testing everything in the API. Thus, I think the commands should be executable in any state and let GNSS return appropriate error if needed. And, it should be easy for the user to understand what a specific error means.

I'm leaning towards keeping the implementation like it is now. Maybe this isn't needed everywhere in mosh, but I think with the GNSS commands it's justified.

Reply:
If you don't make it general, why the error message doesn't say something along the lines written into the header but rather `NRF_EACCES`, which is still quite abstract and you need to check the header. If that's needed, then -13 would be enough for me.

Reply:
Well, **that** is a valid question. Yes, I could probably do that. I just need to check that a particular error code has the same meaning across different GNSS operations.

Reply:
If the meaning is different, I guess we could think of opening the most important ones in a textual format and let some other's be just a number.